core-bundle/src/Controller/BackendPreviewSwitchController.php

```diff
@@ -94,6 +98,20 @@ private function renderToolbar(): string
         $frontendUsername = $this->tokenChecker->getFrontendUsername();
         $showUnpublished = $this->tokenChecker->isPreviewMode();
 
+        $attributes = '';
+
+        if ($this->container->hasParameter('contao.backend.attributes')) {
+            $attributes = $this->container->getParameter('contao.backend.attributes');
+
+            if (!empty($attributes) && \is_array($attributes)) {
+                $attributes = ' '.implode(' ', array_map(
+                    static fn ($v, $k) => sprintf('data-%s="%s"', $k, $v),
```

Comment:
we should probably encode `StringUtil::specialchars` on the value to make sure quotes etc. are correctly encoded.

Reply:
Well, it's just copied from the implementation in the backend template: https://github.com/contao/contao/blob/4fa153799e2a3443c55a1f13d961df79b838fdde/core-bundle/src/Resources/contao/classes/BackendTemplate.php#L182

And there is a validation in the configuration: https://github.com/contao/contao/blob/4fa153799e2a3443c55a1f13d961df79b838fdde/core-bundle/src/DependencyInjection/Configuration.php#L538